src/dodal/devices/i04/murko_results.py

```diff
@@ -98,94 +101,76 @@ async def unstage(self):
-    ) -> np.ndarray | None:
-        """Gets the 'most_likely_click' coordinates (in mm to move the sample) from
-        Murko if omega or the last omega are the closest angle to the search angle.
-        Otherwise returns None.
+        """Uses the 'most_likely_click' coordinates from Murko to calculate the
+        horizontal and vertical distances from the beam centre, and store these values
+        as well as the omega angle the image was taken at.
         """
-        search_angle = self.search_angles[0][0]
-        LOGGER.info(f"Compare {omega}, {search_angle}, {self._last_omega}")
-        if (  # if last omega is closest
-            abs(omega - search_angle) >= abs(self._last_omega - search_angle)
-            and self._last_result is not None
-        ):
-            closest_result = self._last_result
-            closest_omega = self._last_omega
-        elif omega - search_angle >= 0:  # if this omega is closest
-            closest_result = result
-            closest_omega = omega
-        else:
-            return None
-        coords = closest_result[
-            "most_likely_click"
-        ]  # As proportion from top, left of image
-        shape = closest_result["original_shape"]  # Dimensions of image in pixels
+        LOGGER.info(f"Using result {uuid}, {metadata_str}, {result}")
+        metadata = MurkoMetadata(json.loads(metadata_str))
+        omega = metadata["omega_angle"]
+        LOGGER.info(f"Got angle {omega}")
+        coords = result["most_likely_click"]  # As proportion from top, left of image
+        shape = result["original_shape"]  # Dimensions of image in pixels
         # Murko returns coords as y, x
         centre_px = (coords[1] * shape[1], coords[0] * shape[0])
-        LOGGER.info(
-            f"Using image taken at {closest_omega}, which found xtal at {centre_px}"
-        )
+        LOGGER.info(f"Using image taken at {omega}, which found xtal at {centre_px}")
 
         beam_dist_px = calculate_beam_distance(
             (metadata["beam_centre_i"], metadata["beam_centre_j"]),
             centre_px[0],
             centre_px[1],
         )
-
-        return camera_coordinates_to_xyz_mm(
-            beam_dist_px[0],
-            beam_dist_px[1],
-            closest_omega,
-            metadata["microns_per_x_pixel"],
-            metadata["microns_per_y_pixel"],
+        LOGGER.info(f"Found horizontal distance at {beam_dist_px[0]}, angle = {omega}")
+        LOGGER.info(f"Found vertical distance at {beam_dist_px[1]}, angle = {omega}")
+        self.x_dists_mm.append(
+            -beam_dist_px[0] * metadata["microns_per_x_pixel"] / 1000
         )
+        self.y_dists_mm.append(
+            -beam_dist_px[1] * metadata["microns_per_y_pixel"] / 1000
+        )
+        self.omegas.append(omega)
+        self._last_omega = omega
+
+    @staticmethod
+    def get_yz_least_squares(v_values: list, omegas: list) -> tuple[float, float]:
```

Comment:
Could: IMHO, `@staticmethod` isn't very pythonic, you can just use a standalone function outside the class